tests/models_test.py

```diff
@@ -90,3 +93,38 @@ def test_no_top_variable_shape_4(app, last_feature_dim):
     input_shape = (None, None, 4)
     model = app(weights=None, include_top=False, input_shape=input_shape)
     assert model.output_shape == (None, None, None, last_feature_dim)
+
+
+@parametrize
+def test_model_summary(app, last_feature_dim, capsys, snapshot):
+    input_tensor = keras.layers.Input(shape=(224, 224, 3))
+    model = app(weights=None, input_tensor=input_tensor)
+    lq.models.summary(model)
+    out, err = capsys.readouterr()
+
+    summary_file = (
+        Path(__file__).parent
+        / "snapshots"
+        / "model_summaries"
+        / f"{app.__name__}_{last_feature_dim}.txt"
+    )
+
+    if summary_file.exists():
+        with open(summary_file, "r") as file:
+            content = file.read()
+        assert content == out
+    else:
+        with open(summary_file, "w") as file:
+            file.write(out)
```

Comment:
I could be wrong, but does this mean that if the summary file doesn't exist then when the test runs on CI it will write the .txt file (because it's missing) and the test will pass? Maybe we should raise an error here or something after writing to force test failure?

Reply:
Yes, I like that suggestion. I couldn't really think of a way to properly indicate that a new snapshot was generated, but indeed raising an error after writing the summary would be nice. Then if that was intentional, you can just re-run the tests and they'll pass :+1:

Reply:
Yes the warning message can be something along the lines of "Couldn't find snaphot, generating new summary, if this was intentional run the tests again and they will pass"